### ISSUE TITLE
Add Eric's in to out break_match

### ISSUE THREAD

**Issue by Kethku** [uwplse/StructTact#4]
```
Ltac break_match_sm' t :=
 match t with
   | context[match ?X with ​_ => _​ end] =>
     break_match_sm' X || destruct X eqn:?
   | _ => destruct t eqn:?
 end.

Ltac break_match_sm :=
 match goal with
   (* | [ H : context[match ?X with ​_ => _​ end] |- _ ] => *)
   (*   break_match_sm' X *)
   | [ |- context[match ?X with ​_ => _​ end] ] =>
     break_match_sm' X
 end.
```

Ideally there would be a hypothesis and goal version of each of this. There should be a well thought out name for it as well instead of just sm suffix.

**Comment by sliverdragon37** [comment]
I agree, we need a better name than the _sm suffix. The inspiration for that was "break_match the smallest piece", so maybe some name which refers to finding the smallest piece to destruct?

How about "break_inner_match"?

**Comment by dwoos** [comment]
I like `break_inner_match`.

**Comment by sliverdragon37** [comment]
implemented this in #18
